src/ResourceManager/Billing/Commands.Billing/Cmdlets/Invoices/GetAzureRmBillingInvoice.cs

```diff
@@ -33,42 +32,53 @@ public class GetAzureRmBillingInvoice : AzureBillingCmdletBase
 
         [Parameter(Mandatory = true, HelpMessage = "Name of a specific invoice to get.", ParameterSetName = Constants.ParameterSetNames.SingleItemParameterSet)]
         [ValidateNotNullOrEmpty]
-        public string Name { get; set; }
+        public List<string> Name { get; set; }
 
         [Parameter(Mandatory = false, HelpMessage = "Determine the maximum number of records to return.", ParameterSetName = Constants.ParameterSetNames.ListParameterSet)]
         [ValidateNotNull]
+        [ValidateRange(1, 100)]
         public int? MaxCount { get; set; }
 
         [Parameter(Mandatory = false, HelpMessage = "Generate the download url of the invoices.", ParameterSetName = Constants.ParameterSetNames.ListParameterSet)]
         public SwitchParameter GenerateDownloadUrl { get; set; }
 
         public override void ExecuteCmdlet()
         {
-            if (ParameterSetName.Equals(Constants.ParameterSetNames.ListParameterSet))
+            try
             {
-                string expand = this.GenerateDownloadUrl.IsPresent ? DownloadUrlExpand : null;
-
-                if (MaxCount.HasValue && (MaxCount.Value > 100 || MaxCount.Value < 1))
+                if (ParameterSetName.Equals(Constants.ParameterSetNames.ListParameterSet))
                 {
-                    throw new PSArgumentException(Resources.MaxCountExceedRangeError);
-                }
+                    string expand = this.GenerateDownloadUrl.IsPresent ? DownloadUrlExpand : null;
 
-                WriteObject(BillingManagementClient.Invoices.List(expand, null, null, MaxCount).Select(x => new PSInvoice(x)));
-                return;
-            }
+                    WriteObject(BillingManagementClient.Invoices.List(expand, null, null, MaxCount).Select(x => new PSInvoice(x)), true);
+                    return;
+                }
 
-            Invoice invoice = null;
-            if (ParameterSetName.Equals(Constants.ParameterSetNames.LatestItemParameterSet))
-            {
-                invoice = BillingManagementClient.Invoices.GetLatest();
-            }
-            else if (ParameterSetName.Equals(Constants.ParameterSetNames.SingleItemParameterSet))
-            {
-                invoice = BillingManagementClient.Invoices.Get(this.Name);
+                if (ParameterSetName.Equals(Constants.ParameterSetNames.LatestItemParameterSet))
+                {
+                    Invoice invoice = BillingManagementClient.Invoices.GetLatest();
+                    WriteObject(new PSInvoice(invoice));
+                }
+                else if (ParameterSetName.Equals(Constants.ParameterSetNames.SingleItemParameterSet))
+                {
+                    foreach (var invoiceName in Name)
+                    {
+                        try
+                        {
+                            var invoice = new PSInvoice(BillingManagementClient.Invoices.Get(invoiceName));
```

Comment:
Note on optimization: for a future release, you could run asynchronous tasks for each of these, and print the results as each is completed